soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -10886,6 +10906,39 @@ void Player_Update(Actor* thisx, GlobalContext* globalCtx) {
     MREG(53) = this->actor.world.pos.y;
     MREG(54) = this->actor.world.pos.z;
     MREG(55) = this->actor.world.rot.y;
+
+    if (giantLink == 1) {
+        this->actor.scale.x = 0.02f;
+        this->actor.scale.y = 0.02f;
+        this->actor.scale.z = 0.02f;
+    }
+
+    if (minishLink == 1) {
+        this->actor.scale.x = 0.001f;
+        this->actor.scale.y = 0.001f;
+        this->actor.scale.z = 0.001f;
+    }
+
+    if (paperLink != 0) {
+        this->actor.scale.x = paperLink == 1 ? 0.001f : 0.01f;
+        this->actor.scale.y = 0.01f;
+        this->actor.scale.z =  paperLink == 2 ? 0.001f : 0.01f;
+    }
+
+    if (resetLinkScale == 1) {
```

Comment:
if we just used bool instead of uint32_t for these (even if we don't) this could just be `if (resetLinkScale)`